src/util.cpp

```diff
@@ -435,11 +435,116 @@ static void InterpretNegativeSetting(std::string& strKey, std::string& strValue)
     }
 }
 
+/** Internal helper functions for ArgsManager */
+class ArgsManagerHelper {
+public:
+    inline static bool UseDefaultSection(const ArgsManager& am, const std::string& strArg)
```

Comment:
Perhaps add a comment for why we need this function (and another comment lower down next to `m_setSectionOnlyArgs` explaining why those particular arguments shouldn't be applied to non-mainnet configs by default).